---
title: "Bioinfo"
date: 2018-11-27T16:11:43+08:00
draft: false

keywords: ["bioinfo"]
description: "collection of bioinfo analysis"
tags: ["bioinfo"]
categories: ["bioinfo"]
author: "Ming Wang"

# You can also close(false) or open(true) something for this content.
# P.S. comment can only be closed
comment: true
toc: true
autoCollapseToc: false
postMetaInFooter: false
hiddenFromHomePage: false
# You can also define another contentCopyright. e.g. contentCopyright: "This is another copyright."
contentCopyright: false
reward: false
mathjax: false
mathjaxEnableSingleDollar: false

flowchartDiagrams:
  enable: false
  options: ""

sequenceDiagrams: 
  enable: false
  options: ""

---

# Bioinfo analysis

**Mission**

1. `fork` the repo    
2. `pull` the repo to your local machine    
3. update `smRNAseq pipeline` details, `git commit`    
4. `push` your changes to remote repo    

<!--more-->

[Git manual](https://liaoxuefeng.com/wiki/0013739516305929606dd18361248578c67b8067c8c017b000) - Liao Xuefeng 


## smRNAseq pipeline

source: [piPipes_smallRNA](https://github.com/bakerwm/piPipes/blob/master/bin/piPipes_smallRNA.sh)

+ 1. Align to rRNA

```
bowtie -r -S -v -k 1 -p {cpu} --un {not_rRNA} {index} {fq}
```

+ 2. Align to hairpin (unique)

```
bowtie -r --norc -v {mm} -m 1 --best --strata -p {cpu} -S --al {index} --un {not_hp} {index} {fq}
```

+ 3. Align to genome (default)

```
bowtie -r -v {mm} -a --best --strata -p {cpu} -S {index} {fq}
```

+ 4. Length separation

  - siRNA:  
  - piRNA:  
  
  plot length distribution

+ 5. Align to transposon, piRNA clusters

+ 6. Quantification











## General process

Day 1.  
1. Download raw data (65 GB / lane)    
2. Demultiplex (~6 hours)    
3. Demo analysis (~2 hours)    
4. Fastqc and reort (~2 hours) 

Day 2.  
1. Trimming (~2-4 hours)  
2. Alignment, checkout reads content (4~8 hours)  
3. Pipeline analysis (4~10 hours)

Day 3.    
1. Wrap analysis into a html report ()
2. Advance anlysis

Day 4.    
1. Save results to `yulab_cloud`    
2. Discuss, and make changes.  

## General analysis

### 1. Quality Control

[FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) - A quality control tool for high throughput sequence data.

[RSeQC](http://rseqc.sourceforge.net/) - An RNA-seq Quality Control Package

Output: 

+ 1. fastqc output, `html` file.    
+ 2. base quality distribution.  

### 2. Alignment

[STAR]() - 
[bowtie2]() -
[bowtie]() -

`RNA-Seq` : both unique and multiple alignments allowed.  
`CLIPseq` : only `unique` mapped reads.  
`smRNAseq` : only `unique` mapped reads.
`ChIPseq` : only `unique` mapped reads.

Output:

+ 1. Algnment output in `BAM` format.  
+ 2. Alignment command, save to "run.lib".  
+ 3. Alignment log, save to "*.log"  
+ 4. Processed log results, "*.json"

### 3. Quantification

the reads are mapped to transcriptome or genome, and then obtain a gene/transcript level counts.

**Recommend** [featureCounts](http://bioinf.wehi.edu.au/featureCounts/) : much faster, use mulitple threads.  
[HTSeq-count](http://www-huber.embl.de/users/anders/HTSeq/doc/count.html) : one of the most popular counting tool, but it is slow.  
[eXpress](http://cdwscience.blogspot.com/2014/02/mrna-quantification-via-express.html) : slow.

Output:

+ 1. count.txt (matrix)    

### 4. Differential expression

differential expression analysis is carried out by:

[DESeq2](http://bioconductor.org/packages/release/bioc/html/DESeq2.html)   
[EdgeR](http://bioconductor.org/packages/release/bioc/html/edgeR.html)   

Extra Notes
[In RNA-Seq, 2 != 2: Between-sample normalization](https://haroldpimentel.wordpress.com/2014/12/08/in-rna-seq-2-2-between-sample-normalization/)    
[RPKM/FPKM, TPM and raw counts for RNA-seq](http://crazyhottommy.blogspot.com/2015/06/rpkmfpkm-tpm-and-raw-counts-for-rna-seq.html)    
[Youtube video counts vs TPM](https://www.youtube.com/watch?v=ztyjiCCt_lM)    

Output:

+ 1. DE command
+ 2. de gene list  
+ 3. scatter plot  
+ 4. MA plot  
+ 5. volcano plot  
+ 6. PCA plot

### 5. GO and KEGG analysis

[clusterProfiler](https://bioconductor.org/packages/release/bioc/vignettes/clusterProfiler/inst/doc/clusterProfiler.html) : Statistical analysis and visualization of functional profiles for genes and gene clusters    



